From a801af7ec1744ea7b8edc4350e8fb0a23289e9c9 Mon Sep 17 00:00:00 2001
From: Angela Stegmaier <angelabaker@ti.com>
Date: Wed, 7 Aug 2019 09:42:09 -0500
Subject: [PATCH 2/2] waylandsink: Update wldrm NV12 offsets and strides

Stride may not be the same as width, as the wldrm was assuming.
Update the NV12 offset and stride calculation to use the actual
values from the video metadata.

Upstream-Status: Pending

Signed-off-by: Angela Stegmaier <angelabaker@ti.com>
---
 ext/wayland/wldrm.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/ext/wayland/wldrm.c b/ext/wayland/wldrm.c
index 9aca9d5..0b72896 100644
--- a/ext/wayland/wldrm.c
+++ b/ext/wayland/wldrm.c
@@ -38,7 +38,10 @@ gst_wl_drm_memory_construct_wl_buffer (GstMemory * mem, GstWlDisplay * display,
   {
     /* NV12 */
     fourcc = GST_MAKE_FOURCC ('N', 'V', '1', '2');
-    strides[1] = GST_ROUND_UP_4 (video_width);
+    strides[0] = GST_VIDEO_INFO_PLANE_STRIDE (info, 0);
+    strides[1] = GST_VIDEO_INFO_PLANE_STRIDE (info, 1);
+    offsets[0] = GST_VIDEO_INFO_PLANE_OFFSET (info, 0);
+    offsets[1] = GST_VIDEO_INFO_PLANE_OFFSET (info, 1);
   }
   else if(format == GST_VIDEO_FORMAT_I420)
   {
@@ -92,7 +95,7 @@ gst_wl_drm_memory_construct_wl_buffer (GstMemory * mem, GstWlDisplay * display,
     return NULL;
   }
 
-  req2.handle = req1.handle,
+  req2.handle = req1.handle;
 
   ret = drmIoctl(display->fd, DRM_IOCTL_GEM_FLINK, &req2);
   if (ret) {
-- 
1.9.1

